cmake/FindCUDNN.cmake

```diff
@@ -82,6 +82,7 @@ if(CUDNN_INCLUDE_DIR)
     "${CUDNN_VERSION_MAJOR}.${CUDNN_VERSION_MINOR}.${CUDNN_VERSION_PATCH}"
     CACHE
     STRING
+    FORCE
     "cuDNN version"
   )
```

Comment:
According to CMake documentation FORCE is the last option.

> `set(<variable> <value>... CACHE <type> <docstring> [FORCE])`

https://cmake.org/cmake/help/latest/command/set.html

---

To avoid mess - it makes sense to replace STRING->INTERNAL (similar to CUDNN_VERSION_MAJOR/MINOR/PATCH versions).

Reply:
See also this comment: https://github.com/opencv/opencv/pull/19452#issuecomment-773092717

> Based on [standard](https://cmake.org/cmake/help/latest/manual/cmake-developer.7.html):
> CUDNN_VERSION should be moved out of the cache